### PR TITLE
feat(backend): Generate APIs on the backend using openapi

### PR DESCRIPTION
**Pull request by nancymuyeh**
Generate API on the backend using openApiGenerate gradle task #15

### CODE REVIEW

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Good work! Make sure now that the build process depends on the OpenAPI generation

**Comment by nancymuyeh** [comment]
> Good work! Make sure now that the build process depends on the OpenAPI generation

Done

**Review by Koufan-De-King — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -80,4 +82,30 @@ githook {
...
+    useBeanValidation: 'true',
+    performBeanValidation: 'true',
+    interfaceOnly: 'true',
+    useOptional: 'false',
+    useSpringBoot3: 'true',
+    useResponseEntity: 'false'
+  ]
+}
+
+task customBuild {
```
Comment (stephane-segning) [inline build.gradle:109]: Don't use a custom build, but the normal build

**Review by stephane-segning — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 2]
```diff
@@ -80,4 +82,28 @@ githook {
...
+    useBeanValidation: 'true',
+    performBeanValidation: 'true',
+    interfaceOnly: 'true',
+    useOptional: 'false',
+    useSpringBoot3: 'true',
+    useResponseEntity: 'false'
+  ]
+}
+
+tasks.named("openApiGenerate") { dependsOn("build") }
```
Comment (stephane-segning) [inline build.gradle:109]: Good but the opposite. It's the build that should depends on the openapi generation.

**Review by stephane-segning — CHANGES_REQUESTED** [review]

**Review by stephane-segning — APPROVED** [review]

**Review by mbunwe-victor — APPROVED** [review]
Good work

**Comment by stephane-segning** [comment]
The pipeline is failing. This can be due to the fact that the submodule is not downloaded. Try updating the GitHub Workflow file accordingly

**Review by stephane-segning — CHANGES_REQUESTED** [review]
Update the CI to pull the git submodules too

**Comment by nancymuyeh** [comment]
> Update the CI to pull the git submodules too

Done Everything looks good now

**Review by Bansikah — APPROVED** [review]
Great work @nancymuyeh

**Review by stephane-segning — APPROVED** [review]

**Pull request merged**